File: app/src/main/java/com/dante/diary/detail/DiaryCommentsAdapter.java
package com.dante.diary.detail;

import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.dante.diary.R;
import com.dante.diary.interfaces.IOnItemClickListener;
import com.dante.diary.model.Comment;
import com.dante.diary.utils.DateUtil;
import com.jaychang.st.SimpleText;

import java.util.List;

import butterknife.BindView;
import jp.wasabeef.glide.transformations.CropCircleTransformation;

import static com.dante.diary.base.App.context;

/**
 * Created by yons on 17/3/10.
 */

public class DiaryCommentsAdapter extends BaseQuickAdapter<Comment, BaseViewHolder> {

    @BindView(R.id.avatar)
    ImageView avatar;
    @BindView(R.id.userName)
    TextView userName;
    @BindView(R.id.time)
    TextView time;
    @BindView(R.id.content)
    TextView content;

    public DiaryCommentsAdapter(List<Comment> data) {
        super(R.layout.list_comment_item, data);
    }

    public DiaryCommentsAdapter(List<Comment> data, IOnItemClickListener l) {
        super(R.layout.list_comment_item, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, Comment item) {
        helper.addOnClickListener(R.id.commentAvatar)
                .addOnClickListener(R.id.commentName)
                .addOnClickListener(R.id.commentContent);

        TextView commentContent = helper.getView(R.id.commentContent);
        String content = item.getContent();
        if (item.getRecipient() == null) {
            commentContent.setText(content);
        } else {
            //评论内容前面加上"回复 接收人："
            String recipient = item.getRecipient().getName();
            content = String.format("回复 %s：", recipient) + content;

            SimpleText sText = SimpleText.create(helper.itemView.getContext(), content)
                    .first(recipient)
                    .pressedTextColor(R.color.btg_global_text_blue);
            sText.linkify(commentContent);
            commentContent.setText(sText);
        }

        helper.setText(R.id.commentName, item.getUser().getName())
                .setText(R.id.commentTime, DateUtil.getTimeText(item.getCreated()));

        //评论用户的头像
        Glide.with(helper.itemView.getContext())
                .load(item.getUser().getAvatarUrl())
                .bitmapTransform(new CropCircleTransformation(context))
                .into((ImageView) helper.getView(R.id.commentAvatar));
    }
}

File: app/src/main/java/com/example/myapplication/ui/detail/DiaryCommentsAdapter.kt
package com.example.myapplication.ui.detail

import android.widget.ImageView
import androidx.recyclerview.widget.DiffUtil
import com.blankj.utilcode.util.TimeUtils
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.example.myapplication.R
import com.example.myapplication.data.model.Comment

/**
 * @author Du Wenyu
 * 2021/1/8
 */
class DiaryCommentsAdapter : BaseQuickAdapter<Comment, BaseViewHolder>(R.layout.comment_list_item) {

    init {
        setDiffCallback(object : DiffUtil.ItemCallback<Comment>() {
            override fun areItemsTheSame(oldItem: Comment, newItem: Comment): Boolean =
                oldItem.id == newItem.id

            override fun areContentsTheSame(oldItem: Comment, newItem: Comment): Boolean =
                oldItem.content == newItem.content

        })
    }

    override fun convert(holder: BaseViewHolder, item: Comment) {
        holder.setText(R.id.commentName, item.user.name)
            .setText(R.id.commentContent, getFullContent(item))
            .setText(R.id.commentTime, TimeUtils.getFriendlyTimeSpanByNow(item.created.time - 1000))

        val avatar = holder.getView<ImageView>(R.id.commentAvatar)

        Glide.with(context)
            .load(item.user.avatarUrl)
            .transition(DrawableTransitionOptions.withCrossFade())
            .circleCrop()
            .into(avatar)

    }

    private fun getFullContent(item: Comment): String {
        if (item.recipient != null) {
            return context.getString(R.string.reply_to_xxx, item.recipient.name) + item.content
        }
        return item.content
    }


}